texastribune/donations
# Add new blast page

#### What's this PR do?
Adds a new page/form for Blast subscriptions.

#### Why are we doing this? How does it help us?
We wanted to add some new options for the Blast and creating this page at the same time brings the Blast workflow closer in accordance with the rest of the donations pages.

#### How should this be manually tested?
Visit the blast page on [staging](https://donations-testing.herokuapp.com/blast)
For each option (standard, tax-exempt, academic and legislative), for monthly, yearly or session as applicable...
    * create a new donation
    * check for that donation in stripe (customers/transactions tabs)
        * check that the amount and cadence on the item is accurate
    * check that the donation has synced through to salesforce (can look up by stripe subscription or customer id)
    * on the recurring donation object in salesforce, ensure the subscriber email is listed

#### How should this change be communicated to end users?
I'm assuming this part falls to @ashley-hebler and Ryan Kim, but happy to help as needed.

#### Are there any smells or added technical debt to note?
A few things...
* I spent some time (probably more than I should have) trying to get some stripe elements working here to make the front-end's access to products/prices easier, cleaner and more efficient, but without much luck. When I to have more time, I'd really like to get that working
* The new legislative session-only option is set on a two-year cadence in stripe, which means that the user will only get charged every session (since that's on a two-year cadence as well). There are definitely other ways we could do this (it could just be a one-time charge, if that works) but I felt like this was more in line with the spirit of the option on the front-end (240 per session). The only hang-up here is that, to my knowledge, this same cadence isn't available in SF, so we end up on a yearly cadence there, meaning we would need to manually close those opportunities that pop-up in the off year (2026, 2028). I'll reach out to Anna to see if I missed a way of doing this more properly in SF.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recbnhUrrBplaw7Gn?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
